### PR TITLE
Frontend: Nav Search

### LINKED ISSUE #139: Frontend: Nav Search
1. A search bar for the nav.
2. Autocompletish search bar for finding courses and navigating to the course page easily.

### PR DESCRIPTION
**Pull request by DarkPurple141**
* Improved search
* added tags to database
* slightly amended db schema.
* Navbased search

Resolves #139.

### CODE REVIEW

File: frontend/src/App.vue [hunk 1]
```diff
@@ -29,7 +29,7 @@ html, body {
 :root {
   --color-gray: #d5d5d5;
   --color-light-gray: #eaeaea;
-  --color-very-light-gray: #f2f2f2;
+  --color-very-light-gray: #f4f4f4;
```
Comment (NunoDasNeves) [inline frontend/src/App.vue]: You cray

File: frontend/src/components/AppSearch.vue [hunk 2]
```diff
@@ -30,31 +31,33 @@ export default {
...
 }
 </script>
 
-<style scoped>
+<style lang='less' scoped>
 
 .search {
     position: relative;
-    font-size: 0.8em;
+    font-size: 0.7em;
```
Comment (NunoDasNeves) [inline frontend/src/components/AppSearch.vue]: wait i don't get em
it's a proportion of the inherited font size? 0.7em = 70% of parent element's font size?

File: frontend/src/components/reviews-replies/ReviewCard.vue [hunk 3]
```diff
@@ -35,5 +35,5 @@ export default {
 }
 </script>
 
-<style scoped src='../../css/card.less'>
+<style scoped src='../../css/card.less' lang='less'>
```
Comment (NunoDasNeves) [inline frontend/src/components/reviews-replies/ReviewCard.vue]: it's weird how less is called less when it adds more to css

File: frontend/src/views/Course.vue [hunk 4]
```diff
@@ -77,7 +77,19 @@ export default {
     this.$store.dispatch('course/getCourse', this.id)
     this.$store.dispatch('course/getQuestions', this.id)
     this.$store.dispatch('course/getReviews', this.id)
-  }
+  },
+  beforeRouteUpdate ({ params: { id }}, from, next) {
```
Comment (NunoDasNeves) [inline frontend/src/views/Course.vue]: I'm a genius for coming up with this

File: frontend/src/App.vue (hunk 1 above)
Comment (DarkPurple141) [reply frontend/src/App.vue]: The f9 dream lives on.

File: frontend/src/components/AppSearch.vue (hunk 2 above)
Comment (DarkPurple141) [reply frontend/src/components/AppSearch.vue]: It uses 70% of the font-size of the first parent, grandparent... that has a _set_ font-size.

File: frontend/src/components/reviews-replies/ReviewCard.vue (hunk 3 above)
Comment (DarkPurple141) [reply frontend/src/components/reviews-replies/ReviewCard.vue]: 🥇

File: frontend/src/views/Course.vue (hunk 4 above)
Comment (DarkPurple141) [reply frontend/src/views/Course.vue]: 'Something, something and that's why HTC was wrong in the Apple vs HTC case'

**Pull request merged**